internal/policy/verify.go

```diff
@@ -329,6 +329,119 @@ func VerifyRelativeForRef(ctx context.Context, repo *gitinterface.Repository, fi
 	return nil
 }
 
+// VerifyMergeable checks if a change in a featureRef can be merged into the
+// targetRef. For Git reference protection rules, if a rule can be met with one
+// additional signature, this function returns true and without errors to
+// indicate that the merge will meet policy _if_ the resultant RSL entry is
+// signed by an approved party.
+func VerifyMergeable(ctx context.Context, repo *gitinterface.Repository, targetRef, featureRef string) (bool, error) {
+	slog.Debug("Loading policy...")
+	policyState, err := LoadCurrentState(ctx, repo, PolicyRef)
+	if err != nil {
+		return false, err
+	}
+
+	slog.Debug("Loading current set of attestations...")
+	attestationsState, err := attestations.LoadCurrentAttestations(repo)
+	if err != nil {
+		return false, err
+	}
+
+	var fromID gitinterface.Hash
+	slog.Debug(fmt.Sprintf("Identifying latest RSL entry for '%s'...", targetRef))
+	targetEntry, _, err := rsl.GetLatestReferenceEntry(repo, rsl.ForReference(targetRef), rsl.IsUnskipped())
```

Comment:
> @wlynch: Why resolve the target ids via the RSL instead of via the refs directly?
>
> Could the RSL potentially be behind the ref value if a new commit was pushed without updating the RSL?

Possibly, yes. This is consistent with everywhere else we need this info. For example, for an approval attestation, we also look at the latest RSL entry for the feature branch from which we're approving a change. Should we track this separately?

Reply:
Filed https://github.com/gittuf/gittuf/issues/668